### PR TITLE
add bootstrapping power and nPlan

### PR DESCRIPTION
**Pull request by rosanneturner**

### CODE REVIEW

File: R/safe2x2Test.R [hunk 1]
```diff
@@ -1550,19 +1556,50 @@ simulateWorstCaseQuantileTwoProportions <- function(na, nb, priorValues,
...
       currentWorstCaseQuantile <- currentQuantile
+      if (expectedStopTime) {
+        bootResult <- boot::boot(
+          data = stoppingTimes,
+          statistic = function(x, idx) {
+            mean(x[idx])
+          },
+          R = bootN
+        )
+      } else {
```
Comment (AlexanderLyNL) [inline R/safe2x2Test.R:1573]: I'll write a generic wrapper for this. Add a todo here so it's easier to swap. I'm not sure why the bootstrap is run while you run through the loop though.

File: R/safe2x2Test.R [hunk 2]
```diff
@@ -1550,19 +1556,50 @@ simulateWorstCaseQuantileTwoProportions <- function(na, nb, priorValues,
...
     if (currentPower <= currentWorstCasePower) {
       currentWorstCasePower <- currentPower
+      bootResultPower <- boot::boot(
+        data = stopEs,
+        statistic = function(x, idx) {
+          mean(x[idx] >= 1/alpha)
+        },
+        R = bootN
+      )
+      powerTwoSe <- sd(bootResultPower[["t"]])
```
Comment (AlexanderLyNL) [inline R/safe2x2Test.R:1593]: two times?

File: R/safe2x2Test.R [hunk 3]
```diff
@@ -1550,19 +1556,50 @@ simulateWorstCaseQuantileTwoProportions <- function(na, nb, priorValues,
...
+            quantile(x[idx], 1 - beta)
+          },
+          R = bootN
+        )
+      }
+      worstCaseQuantileTwoSe <- sd(bootResult[["t"]])
     }
     if (currentPower <= currentWorstCasePower) {
       currentWorstCasePower <- currentPower
+      bootResultPower <- boot::boot(
```
Comment (AlexanderLyNL) [inline R/safe2x2Test.R:1586]: I'll write a generic wrapper for this. Add a todo here so it's easier to swap. I'm not sure why the bootstrap is run while you run through the loop though.

File: R/safe2x2Test.R [hunk 4]
```diff
@@ -124,16 +124,18 @@ designSafeTwoProportions <- function(na, nb,
   }
 
   names(priorValuesForPrint) <- "Beta hyperparameters"
-  impliedTarget <- NULL
+  impliedTarget <- nPlanTwoSe <- betaTwoSe <- NULL
```
Comment (AlexanderLyNL) [inline R/safe2x2Test.R:127]: logImpliedTarget <- logImpliedTargetTwoSe <-

File: R/safe2x2Test.R (hunk 2 above)
Comment (rosanneturner) [reply R/safe2x2Test.R]: je hebt helemaal gelijk, dit deed ik nu in de print functie pas, dan klopt er niets van deze variabelenaam

File: R/safe2x2Test.R (hunk 1 above)
Comment (rosanneturner) [reply R/safe2x2Test.R:1576]: we finally report the nPlan for the worst case distribution in H1(delta), so need to store that one and their SE. Could store the stopping times and E variables for worst distribution until end of loop. Have added todo, let's do this together with the beautifying in the next iteration of the code. If we do not make simulation number M too high this won't take up too much memory.

File: R/safe2x2Test.R (hunk 4 above)
Comment (rosanneturner) [reply R/safe2x2Test.R:127]: added todo!

**Pull request merged**